driver-core/src/main/com/mongodb/client/model/CreateCollectionOptions.java

```diff
@@ -360,4 +361,19 @@ public String toString() {
                 + ", encryptedFields=" + encryptedFields
                 + '}';
     }
+
+    /**
+     * Creates a shallow copy of {@code this} {@link CreateCollectionOptions} by calling {@code super.clone()}.
+     *
+     * @return A shallow copy of {@code this} {@link CreateCollectionOptions}.
+     * @since 4.9
+     */
+    @Override
+    public CreateCollectionOptions clone() {
+        try {
+            return (CreateCollectionOptions) super.clone();
+        } catch (CloneNotSupportedException e) {
+            throw fail(e.toString());
+        }
+    }
```

Comment:
1. Implementing `Cloneable` on `CreateCollectionOptions` is backward-compatible. Also, this implementation does not screw over classes that extend `CreateCollectionOptions`: they can override `clone` by calling `super.clone`, cast to their type and it will work.
2. I hesitated between creating a copy constructor and adding `clone`. While I prefer copy constructors, `clone` guarantees that we won't forget to copy new fields in the copy constructor when we add more fields.

I am not full of confidence here and is open to hearing opinions if there are any.

Reply:
Done.